client/galaxy/scripts/layout/menu.js

```diff
@@ -12,6 +12,56 @@ var Collection = Backbone.Collection.extend({
         this.reset();
 
         //
+        // Chat server tab
+        //
+        this.add({
+            id              : 'show-chat-online',
+            icon            : 'fa-comment-o',
+            tooltip         : 'Chat Online',
+            visible         : false,
+            onclick         : function( e ) {
+                // make modal
+                var host =  window.Galaxy.config.communication_server_host,
+                    port = window.Galaxy.config.communication_server_port,
+                    username = escape(window.Galaxy.user.attributes.username),
+                    persistent_communication_rooms = escape(window.Galaxy.config.persistent_communication_rooms),
+                    query_string = "?username=" + username + "&persistent_communication_rooms=" + persistent_communication_rooms,
+                    src = host + ":" + port + query_string,
+                    $el_chat_modal_header = null,
+                    $el_chat_modal_body = null,
+                    header_template = "",
+                    template = "<iframe class='f-iframe fade in' src=" + src + " style='width:100%; height:100%;'> </iframe>";
+                // deletes the chat modal if already present and create one
+                if( $( '.modal-dialog' ).find( 'iframe' ).parent().parent().parent().length > 0 ) {
```

Comment:
Traversing the DOM like this is usually susceptible to errors - could you use some more stable selector?